Applications/Spire/Include/Spire/Ui/EditableTableView.hpp

```diff
@@ -0,0 +1,114 @@
+#ifndef SPIRE_EDITABLE_TABLE_VIEW_HPP
+#define SPIRE_EDITABLE_TABLE_VIEW_HPP
+#include "Spire/Spire/ArrayListModel.hpp"
+#include "Spire/Spire/FilteredTableModel.hpp"
+#include "Spire/Ui/TableView.hpp"
+
+namespace Spire {
+
+  /**
+   * Customizes the PopupBox to propagate mouse events to TableItem to
+   * trigger the current signal for TableView.
+   */
+  class TransparentMouseEventsPopupBox : public QWidget {
+    public:
+
+      /** Constructs a TransparentMouseEventsPopupBox. */
+      explicit TransparentMouseEventsPopupBox(QWidget& body,
+        QWidget* parent = nullptr);
+
+    protected:
+      bool eventFilter(QObject* watched, QEvent* event) override;
+      bool event(QEvent* event) override;
+      void showEvent(QShowEvent* event) override;
+      void keyPressEvent(QKeyEvent* event) override;
+
+    private:
+      PopupBox* m_popup_box;
+      QWidget* m_tip_window;
+      bool m_has_sent_event;
+  };
+
+  /** Represents an editable TableView. */
+  class EditableTableView : public TableView {
+    public:
+
+      using HeaderModel = TableView::HeaderModel;
+
+      using ViewBuilder = TableView::ViewBuilder;
+
+      using Index = TableView::Index;
+
+      using CurrentModel = TableView::CurrentModel;
+
+      using SelectionModel = TableView::SelectionModel;
+
+      using Comparator = TableView::Comparator;
+
+      using Filter = FilteredTableModel::Filter;
+
+      /**
+       * Signals that the row is being clicked to delete.
+       */
+      using DeleteSignal = Signal<void (int row)>;
+
+      /**
+       * Constructs an EditableTableView.
+       * @param table The model of values to display.
+       * @param header The model used to display the header.
+       * @param table_filter The filter to apply to a column.
+       * @param current The current value.
+       * @param selection The selection.
+       * @param view_builder The ViewBuilder to use.
+       * @param comparator The comparison function.
+       * @param filter The filter to apply to the entire table.
+       * @param parent The parent widget.
+       */
+      EditableTableView(std::shared_ptr<TableModel> table,
+        std::shared_ptr<HeaderModel> header,
+        std::shared_ptr<TableFilter> table_filter,
+        std::shared_ptr<CurrentModel> current,
+        std::shared_ptr<SelectionModel> selection, ViewBuilder view_builder,
+        Comparator comparator, Filter filter, QWidget* parent = nullptr);
+
+      /** Applies a new filter to this TableView. */
+      void set_filter(const Filter& filter);
```

Comment:
I don't see a purpose for this method, changes to the model should be handled by the ```TableModel```.